File: doc-examples/jdbc-example-java/src/main/java/example/QuantityTypeConverter.java

package example;

import io.micronaut.core.convert.ConversionContext;
import io.micronaut.data.model.runtime.convert.TypeConverter;
import jakarta.inject.Singleton;

@Singleton // <1>
public class QuantityTypeConverter implements TypeConverter<Quantity, Integer> {

    @Override // <2>
    public Integer convertToPersistedValue(Quantity quantity, ConversionContext context) {
        return quantity == null ? null : quantity.getAmount();
    }

    @Override // <3>
    public Quantity convertToEntityValue(Integer value, ConversionContext context) {
        return value == null ? null : Quantity.valueOf(value);
    }

}

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/QuantityTypeConverter.kt
package example

import io.micronaut.core.convert.ConversionContext
import io.micronaut.data.model.runtime.convert.TypeConverter
import jakarta.inject.Singleton

@Singleton // <1>
class QuantityTypeConverter : TypeConverter<Quantity?, Int?> {

    // <2>
    override fun convertToPersistedValue(quantity: Quantity?, context: ConversionContext): Int? {
        return quantity?.amount
    }

    // <3>
    override fun convertToEntityValue(value: Int?, context: ConversionContext): Quantity? {
        return if (value == null) null else Quantity(value)
    }

}